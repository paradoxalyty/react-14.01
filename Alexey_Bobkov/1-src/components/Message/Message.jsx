import React from 'react';
import PropTypes from 'prop-types';
import './Message.sass';
import classnames from 'classnames';
import { CircularProgress } from '@material-ui/core';

export const Message = ({ name, content, isLoading }) => {

    const classNames = classnames('Message', { 'Message Robot': name === 'Robot' });

    if (isLoading === true) {
        return <CircularProgress />
    }

    return (
        (
            <div className={classNames}>
                <strong>{name}: </strong>
                {content}
            </div>)
    );

}

Message.propTypes = {
    name: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired,


};