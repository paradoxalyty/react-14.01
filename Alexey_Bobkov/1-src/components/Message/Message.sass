.Message
    background-color: #6fa9eb
    margin: 5px 15px 5px 15px
    padding: 10px
    align-self: flex-end
    border-radius: 3px
    
.Robot
    align-self: flex-start
.MuiCircularProgress-root
    align-self: center
    margin-top: 10%
    
.MuiCircularProgress-colorPrimary
    color: red !important