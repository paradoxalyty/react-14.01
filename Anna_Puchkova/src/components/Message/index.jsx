import React from "react";
import PropTypes from 'prop-types';
import cn from 'classnames';
import '../../styles/styles.css';

export default class Message extends React.Component {
    static propTypes = {
        text: PropTypes.string.isRequired,
        sender: PropTypes.string.isRequired
    };
 
    render() {
        const robotMessage = cn({
            'message': true,
            'messageRobot': this.props.sender === 'bot',
          });

        return <div className={robotMessage}>
                    <div>{ this.props.text }</div>
                    <div className="message-sender">{ this.props.sender }</div>            
               </div>

    }
 }
 

