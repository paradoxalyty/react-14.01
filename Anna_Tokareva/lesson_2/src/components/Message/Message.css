.Message {
  position: relative;
  background-color: lightskyblue;
  padding: 5px;
  border-radius: 5px;
  width: 30%;
  align-self: flex-start;
  overflow-wrap: break-word;
}

.Message:not(:last-child) {
  margin-bottom: 20px;
}

.Message--user {
  align-self: flex-end;
}

.DeleteIcon {
  position: absolute;
  top: 1px;
  right: 1px;
  display: none;
}

.DeleteIcon--user {
  display: block;
}
