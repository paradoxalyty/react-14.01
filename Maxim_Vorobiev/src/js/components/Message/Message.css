.react-messenger__message {
    background-color: lightcoral;
    padding: 5px 15px;
    border-radius: 20px;
    font-size: 18px;
    margin: 5px;
    align-self: flex-start;
}

.react-messenger__message-bot {
    align-self: flex-end;
}

.react-messenger__message-name {
    color: lightgray;
    font-size: 14px;
}

