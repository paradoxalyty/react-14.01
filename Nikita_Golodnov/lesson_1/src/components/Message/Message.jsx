import React from 'react';
import PropTypes from 'prop-types';
// import './Message.css';
import classnames from 'classnames';

export const Message = ({text, content}) => {
    const classNames = classnames ('Message', {'Message__bot': text === 'Bot'})
    return (<div className={classNames}><strong>{text}</strong><br/>{content}</div>)
}


Message.propTypes = {
    text: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired 
}
 

