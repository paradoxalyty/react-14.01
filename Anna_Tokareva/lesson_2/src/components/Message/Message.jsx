import React, { useState } from "react";
import PropTypes from "prop-types";
import "./Message.css";
import classnames from "classnames";
import { makeStyles } from "@material-ui/core/styles";
import IconButton from "@material-ui/core/IconButton";
import DeleteIcon from "@material-ui/icons/Delete";

const useStyles = makeStyles(theme => ({
  root: {
    "& > *": {
      margin: theme.spacing(1)
    }
  }
}));

export const Message = ({ name, content, messageId, deleteMessage }) => {
  const classes = useStyles();
  const [del, setDel] = useState(false);
  const classNames = classnames("Message", {
    "Message--user": name === "Я"
  });
  const classNamesDeleteIcon = classnames("DeleteIcon", {
    "DeleteIcon--user": name === "Я" && del
  });

  const handleDeleteClik = messageId => {
    deleteMessage(messageId);
  };

  return (
    <div
      onMouseOver={() => setDel(true)}
      onMouseLeave={() => setDel(false)}
      className={classNames}
    >
      <strong>{name}: </strong>
      <div>{content}</div>
      <span className={classNamesDeleteIcon}>
        <DeleteIcon
          aria-label="delete"
          disabled
          color="primary"
          onClick={() => handleDeleteClik(messageId)}
        />
      </span>
    </div>
  );
};

Message.propTypes = {
  name: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired,
  messageId: PropTypes.string.isRequired,
  deleteMessage: PropTypes.func.isRequired
};
