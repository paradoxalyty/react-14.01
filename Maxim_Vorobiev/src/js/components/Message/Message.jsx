import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import './Message.css';

const BOT_NAME = 'Bot';

export const Message = ({name, content}) => {
    const classNames = classnames('react-messenger__message', {'react-messenger__message-bot': name === BOT_NAME});

    return (
        <div className={classNames}>
            <div>{content}</div>

            <div className="react-messenger__message-name">
                {name}
            </div>
        </div>
    )
};

Message.propTypes = {
    name: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired,
};

Message.defaultProps = {
    name: 'Name',
    content: "Message",
};