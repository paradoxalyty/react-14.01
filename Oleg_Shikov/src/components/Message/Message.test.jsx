import React from 'react';
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer'
import {Message} from './Message';

describe("Message", () => {
    it("contains name", ()=> {
        const props = {
            name: "Ivan",
            content: "Hello",
        }
        const element = shallow(<Message {...props} />);

        expect(element.contains(<strong>{props.name}:</strong>)).toBe(true)

    });

    it("render content", ()=> {
        const props = {
            name: "Ivan",
            content: "Hello",
        }
        const element = renderer.create(<Message {...props} />).toJSON();

        expect(element).toMatchSnapshot();

    });
});