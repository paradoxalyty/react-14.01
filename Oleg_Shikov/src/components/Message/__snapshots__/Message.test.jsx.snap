// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message render content 1`] = `
<div
  className="Message"
>
  <strong>
    Ivan
    :
  </strong>
   
  Hello
</div>
`;
